presto-common/src/main/java/com/facebook/presto/common/plan/PlanCanonicalizationStrategy.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.plan;
+
+public enum PlanCanonicalizationStrategy
+{
+    /**
+     * DEFAULT strategy is used to canonicalize plans with minimal changes.
+     *
+     * We remove any unimportant information like source location, make the variable
+     * names consistent, and order them.
+     * For example:
+     * `SELECT * FROM table WHERE id IN (1, 2)` will be equivalent to `SELECT * FROM table WHERE id IN (2, 1)`
+     *
+     * This is used in context of fragment result caching
+     */
+    DEFAULT,
+    /**
+     * CONNECTOR strategy will canonicalize plan according to DEFAULT strategy, and additionally
+     * canoncialize `TableScanNode` by giving a connector specific implementation.
+     *
+     * With this approach, we call ConnectorTableLayoutHandle.getIdentifier() for all `TableScanNode`.
+     * Each connector can have a specific implementation to canonicalize table layout handles however they want.
+     *
+     * For example, Hive connector removes constants from constraints on partition keys:
+     * `SELECT * FROM table WHERE ds = '2020-01-01'` will be equivalent to `SELECT * FROM table WHERE ds = '2020-01-02'`
+     * where `ds` is a partition column in `table`
+     *
+     * This is used in context of history based optimizations.
+     */
+    CONNECTOR,
```

Comment:
Is this still needed? Looks like it is only used in tests.

Reply:
The original plan was to have remove_constants approach as fallback approach. Next PR still has this fallback logic I'd like to keep the fallback approach and compare 2 strategies. This could still be useful if we have weird query patterns like:

`select * from A join (select 1 as id) B ON A.id = B.id`
vs
`select * from A join (select 100 as id) B ON A.id = B.id`

final statistics can change based on how skewed A is